src/tools/illink/src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1954,14 +1954,14 @@ internal void MarkStaticConstructorVisibleToReflection (TypeDefinition type, in
 			MarkStaticConstructor (type, reason, origin);
 		}
 
-
 		/// <summary>
 		/// Marks the specified <paramref name="reference"/> as referenced.
 		/// </summary>
 		/// <param name="reference">The type reference to mark.</param>
 		/// <param name="reason">The reason why the marking is occuring</param>
+		/// <param name="addToGraphAsRoot">Whether to add the node to the dependency graph</param>
 		/// <returns>The resolved type definition if the reference can be resolved</returns>
-		protected internal virtual TypeDefinition? MarkType (TypeReference reference, DependencyInfo reason, MessageOrigin? origin = null)
+		protected internal virtual TypeDefinition? MarkType (TypeReference reference, DependencyInfo reason, MessageOrigin? origin = null, bool addToGraphAsRoot = true)
```

Comment:
The additional parameter is just a bandaid to make this work until a follow up separates out the resolving, premarking, and marking correctly. I just don't want to make this PR larger than it needs to be.